server/src/services/apps.service.ts

```diff
@@ -294,6 +294,14 @@ export class AppsService {
       where: { id: versionFromId },
     });
 
+    const isVersionNameExisted = await this.appVersionsRepository.findOne({
```

Comment:
When concurrent requests are served, there is a chance that this check passes. The DB would still throw an error due to a unique constraint that is now kept in. 

Currently, the server wouldn't handle it thus returning an unintelligible error to the user with 500 status. This needs to be handled. That can be a separate PR. Probably we can configure typeorm to handle such DB errors. This will in turn make our code easier and we wouldn't have to do these checks manually to raise a custom error message.